Mathlib/AlgebraicGeometry/Modules/Tilde.lean

```diff
@@ -252,6 +254,142 @@ noncomputable def localizationToStalk (x : PrimeSpectrum.Top R) :
+    (hx : x ∈ U) :
+    ∃ (V : Opens (PrimeSpectrum.Top R)) (_ : x ∈ V) (i : V ⟶ U) (f : M) (g : R) (hg : _),
+      const M f g V hg = (tildeInModuleCat M).map i.op s :=
+  let ⟨V, hxV, iVU, f, g, hfg⟩ := s.2 ⟨x, hx⟩
+  ⟨V, hxV, iVU, f, g, fun y hyV => (hfg ⟨y, hyV⟩).1, Subtype.eq <| funext fun y => by
+    obtain ⟨h1, (h2 : g • s.1 ⟨y, _⟩ = LocalizedModule.mk f 1)⟩ := hfg y
+    exact show LocalizedModule.mk f ⟨g, by exact h1⟩ = s.1 (iVU y) by
+      set x := s.1 (iVU y); change g • x = _ at h2; clear_value x
+      induction x using LocalizedModule.induction_on with
+      | h a b =>
+        rw [LocalizedModule.smul'_mk, LocalizedModule.mk_eq] at h2
+        obtain ⟨c, hc⟩ := h2
+        exact LocalizedModule.mk_eq.mpr ⟨c, by simpa using hc.symm⟩⟩
+
+
+@[simp]
+theorem res_const (f : M) (g : R) (U hu V hv i) :
+    (tildeInModuleCat M).map i (const M f g U hu) = const M f g V hv :=
+  rfl
+
+@[simp]
+theorem localizationToStalk_mk (x : PrimeSpectrum.Top R) (f : M) (s : x.asIdeal.primeCompl) :
+    localizationToStalk M x (LocalizedModule.mk f s) =
+      (tildeInModuleCat M).germ (PrimeSpectrum.basicOpen (s : R)) x s.2
+        (const M f s (PrimeSpectrum.basicOpen s) fun _ => id) :=
+  (Module.End_isUnit_iff _ |>.1 (isUnit_toStalk M x s)).injective <| by
+  erw [← LinearMap.mul_apply]
+  simp only [IsUnit.mul_val_inv, LinearMap.one_apply, Module.algebraMap_end_apply]
+  show (M.tildeInModuleCat.germ ⊤ x ⟨⟩) ((toOpen M ⊤) f) = _
+  rw [← map_smul]
+  fapply TopCat.Presheaf.germ_ext (W := PrimeSpectrum.basicOpen s.1) (hxW := s.2)
+  · exact homOfLE le_top
+  · exact 𝟙 _
+  refine Subtype.eq <| funext fun y => show LocalizedModule.mk f 1 = _ from ?_
+  show LocalizedModule.mk f 1 = s.1 • LocalizedModule.mk f _
+  rw [LocalizedModule.smul'_mk, LocalizedModule.mk_eq]
+  exact ⟨1, by simp⟩
+
+/--
+The isomorphism of `R`-modules from the stalk of `M^~` at `x` to the localization of `M` at the
+prime ideal corresponding to `x`.
+-/
+@[simps]
+noncomputable def stalkIso (x : PrimeSpectrum.Top R) :
+    TopCat.Presheaf.stalk (tildeInModuleCat M) x ≅
+    ModuleCat.of R (LocalizedModule x.asIdeal.primeCompl M) where
+  hom := stalkToFiberLinearMap M x
+  inv := localizationToStalk M x
+  hom_inv_id := TopCat.Presheaf.stalk_hom_ext _ fun U hxU ↦ ext _ fun s ↦ by
+    show localizationToStalk M x (stalkToFiberLinearMap M x (M.tildeInModuleCat.germ U x hxU s)) =
+      M.tildeInModuleCat.germ U x hxU s
+    rw [stalkToFiberLinearMap_germ]
+    obtain ⟨V, hxV, iVU, f, g, (hg : V ≤ PrimeSpectrum.basicOpen _), hs⟩ :=
+      exists_const _ _ s x hxU
+    rw [← res_apply M U V iVU s ⟨x, hxV⟩, ← hs, const_apply, localizationToStalk_mk]
+    exact (tildeInModuleCat M).germ_ext V hxV (homOfLE hg) iVU <| hs ▸ rfl
+  inv_hom_id := by ext x; exact x.induction_on (fun _ _ => by simp)
+
+instance (x : PrimeSpectrum.Top R) : IsLocalizedModule x.asIdeal.primeCompl (stalkIso M x).hom :=
```

Comment:
No this means `M_x` is the localization of `M_x` at `x` right?

Reply:
Indeed

Reply:
now it should be okay